src/components/Homepage/HomePage.jsx

```diff
@@ -1,61 +1,105 @@
 import React, { useEffect } from "react";
 import { useState } from "react";
-import { Link } from "react-router-dom";
-const axios = require("axios");
+import { Link, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
+import Spinner from "../Spinner/Spinner";
 
 const api_key = process.env.REACT_APP_API_KEY;
 
-export default function HomePage({ setMovieInfo }) {
+export default function HomePage({ isLoading, setIsLoading }) {
   const [movies, setMovies] = useState([]);
+  const [totalCountPages, setTotalCountPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const navigate = useNavigate();
+  const params = useParams();
 
   useEffect(() => {
     fetchData();
+    fetchPageData(params.page);
   }, []);
 
-  async function fetchMovieData(id) {
-    const response = await axios.get(
-      `https://api.themoviedb.org/3/movie/${id}`,
-      {
-        params: {
-          api_key: api_key,
-        },
-      }
-    );
-    console.log(response);
-    setMovieInfo(response.data);
+  async function fetchData() {
```

Comment:
Please open network tab and see that every time you open the home page you trigger 2 same requests to the API. This is an incorrect behaviour. We need to avoid unnecessary API calls.

Reply:
fixed